domain-server/src/DomainServerAcmeClient.cpp

```diff
@@ -0,0 +1,751 @@
+    AcmeHttpChallengeFiles(std::map<std::string, std::string> dirs) :
+        dirs(std::move(dirs))
+    {}
+
+    ~AcmeHttpChallengeFiles() override {
+        std::set<QString> challengeDirs;
+
+        for(auto&& challengePath : challengePaths) {
+            challengeDirs.insert(QFileInfo(challengePath).path());
+            if(!QFile(challengePath).remove()) {
+                qCWarning(acme_client) << "Failed to remove challenge file:" << challengePath;
+            }
+        }
+
+        for(auto&& challengeDir : challengeDirs) {
+            if(!QDir(challengeDir).rmdir(".")) {
+                qCWarning(acme_client) << "Failed to remove challenge directory:" << challengeDir;
+            }
+        }
+    }
+
+    std::chrono::milliseconds selfCheckDuration() override { return 2s; }
+    std::chrono::milliseconds selfCheckInterval() override { return 250ms; }
+
+    void addChallenge(const std::string& domain, const std::string& location, const std::string& content) override {
+        QString challengePath = (dirs[domain] + location).c_str();
+        QDir challengeDir = QFileInfo(challengePath).path();
+        if(challengeDir.mkpath(".") && writeAll(content, challengePath)) {
+            challengePaths.push_back(challengePath);
+        } else {
+            qCCritical(acme_client) << "Failed to write challenge file:" << challengePath;
+        }
+    }
+
+private:
+    std::map<std::string, std::string> dirs;
+    std::vector<QString> challengePaths;
+};
+
+class AcmeHttpChallengeManual : public AcmeChallengeHandler {
+public:
+    void addChallenge(const std::string& domain, const std::string& location, const std::string& content) override {
+        qCDebug(acme_client) << "Please manually complete this http challenge:\n"
+            << "Domain:" << domain.c_str() << '\n'
+            << "Location:" << location.c_str() << '\n'
+            << "Content:" << content.c_str() << '\n';
+    }
+
+    std::chrono::milliseconds selfCheckDuration() override { return 120s; }
+    std::chrono::milliseconds selfCheckInterval() override { return 1s; }
+
+};
+
+struct ChallengeHandlerParams {
+    std::string typeId;
+    std::map<std::string, std::string> domainDirs;
+};
+
+std::unique_ptr<AcmeChallengeHandler> makeChallengeHandler(ChallengeHandlerParams params) {
+
+    if(params.typeId == "server") {
+        return std::make_unique<AcmeHttpChallengeServer>();
+    } else if(params.typeId == "files") {
+        return std::make_unique<AcmeHttpChallengeFiles>(params.domainDirs);
+    } else if(params.typeId == "manual") {
+        return std::make_unique<AcmeHttpChallengeManual>();
+    } else {
+        throw std::logic_error("Invalid ACME HTTP challenge handler type id: " + params.typeId);
+    }
+};
+
+template <typename Callback>
+class ChallengeSelfCheck :
+    public std::enable_shared_from_this< ChallengeSelfCheck<Callback> >
+{
+    public:
+    ChallengeSelfCheck(Callback callback) :
+        callback(std::move(callback))
+    {}
```

Comment:
Indenting of "public:" and definitions not consistent with rest of codebase.
Similarly for "private:".